parser/model/model.go

```diff
@@ -234,6 +234,12 @@ const ExtraHandleID = -1
 // ExtraPidColID is the column ID of column which store the partitionID decoded in global index values.
 const ExtraPidColID = -2
 
+// ExtraPhysTblID is the column ID of column that should be filled in with the physical table id.
+// Primarily used for table partition dynamic prune mode, to return which partition (physical table id) the row came from.
+// Using a dedicated id for this, since in the future ExtraPidColID and ExtraPhysTblID may be used for the same request.
+// Must be after ExtraPidColID!
+const ExtraPhysTblID = -3
```

Comment:
Logically, are `ExtraPhysTblID` and `ExtraPidColID` always the same (since `PartitionID` is just another name of the physical table ID of the corresponding partition)? While in implementation, they have different implementations.
Do I understand it correctly? If it is, could we remove `ExtraPidColID` in the future?

Reply:
They will return the same information, but will be read from different sources, `ExtraPhysTblID` extracts the table id from the key (which is in the key-prefix for both table and index data) while `ExtraPidColID` is a normal column stored in a global index (with the logical table id in the key-prefix of the index key).

I think we should be able to remove the `ExtraPidColID` from TiKV and treat it as a hidden column in TiDB, which will only be used in the case of global index, but I think we can still use the dedicated `ExtraPidColID` for partition id/physical table id as a hidden column.

In theory we could extend the global index implementation to support partitioned indexes with different partitioning from the table, then we would need to use both `ExtraPhysTblID` for which index partition the row was read, and `ExtraPidColID` for which table partition the rowID/handle/PK belongs to.

Reply:
OK, got it.